tests/integration/block_createBlock_api_test.go

```diff
@@ -43,27 +43,37 @@ func TestBlockCreateAPI(t *testing.T) {
 type BlockImageCreateSuite struct {
 	suite.Suite
 	testClient     *clients.TestClient
+	kh             *utils.K8sHelper
 	rc             contracts.RestAPIOperator
 	initBlockCount int
 	installer      *installer.InstallHelper
 }
 
 func (s *BlockImageCreateSuite) SetupSuite() {
+	var err error
 
-	kh, err := utils.CreateK8sHelper(s.T)
-	require.NoError(s.T(), err)
+	s.kh, err = utils.CreateK8sHelper(s.T)
+	assert.NoError(s.T(), err)
 
-	s.installer = installer.NewK8sRookhelper(kh.Clientset, s.T)
+	s.installer = installer.NewK8sRookhelper(s.kh.Clientset, s.T)
 
-	err = s.installer.InstallRookOnK8s("rook")
-	require.NoError(s.T(), err)
+	isRookInstalled, err := s.installer.InstallRookOnK8s("rook")
+	if !isRookInstalled {
+		logger.Errorf("Rook Was not installed successfully")
+		s.TearDownSuite()
+		s.T().FailNow()
+	}
 
-	s.testClient, err = clients.CreateTestClient(enums.Kubernetes, kh, "rook")
-	require.Nil(s.T(), err)
+	s.testClient, err = clients.CreateTestClient(s.kh, "rook")
+	if err != nil {
+		logger.Errorf("Cannot create rook test client, er -> %v", err)
+		s.TearDownSuite()
```

Comment:
the tear down isn't implicit?

Reply:
it's not when there is a failure in SetUp, the teardown is never called if set up fails.